algosdk/testing/dryrun.py

```diff
@@ -485,23 +522,33 @@ def _checked_request(
         """
         drr = self._dryrun_request(prog_drr_txns, lsig, app, sender)
         if drr["error"]:
-            self._fail(f"error in dryrun response: {drr['error']}")
+            _fail(f"error in dryrun response: {drr['error']}")
 
         if not drr["txns"]:
-            self._fail("empty response from dryrun")
+            _fail("empty response from dryrun")
 
         return drr["txns"]
 
-    def _fail(self, msg):
-        try:
-            self.fail(msg)
-        except AttributeError:
-            raise AssertionError(msg)
-
 
 class Helper:
     """Utility functions for dryrun"""
 
+    @classmethod
+    def build_simple_logicsig_request(
```

Comment:
@tzaffi Optional - Feel welcomed to close if you prefer as is.

I typically prefer to avoid including the term _simple_ in method names.  In my experience, the convention creates confusion when alternative factories appear in the future.

Alternatively, I'd name it according to its functionality _like_ `build_dryrun_request_for_logicsig` to follow the convention of `build_dryrun_request`.

Reply:
👍

Reply:
I wasn't sure about the wording, but I wanted to make it clear that there is something _simpler_ about this request. I've gone with the following for now. Let me know your thoughts:
* `singleton_logicsig_request()`
* `singleton_app_request)`
Though I'm dropping `dryrun` from the name, in context it should still be clear (if you import the library). EG:
```python
from algosdk.testing.dryrun import Helper as DryRunHelper

...
req = DryRunHelper.singleton_logicsig_request(...)
```

Reply:
Resol.ving this zombie conversation. "simple" nomenclature has long been removed

Reply:
@tzaffi I think I misunderstood the scope of _simple_ on my initial review.  I focused on the app vs logic specification rather than what I think you intended:  `List[bytes]` specification.

Having made another pass, here's my thinking:
* I realize `build_dryrun_request` predates the PR.  I think it's surprising the method supports invocation _without_ specifying execution mode (lsig vs app).  What do folks think about requiring an execution mode? 
  * It's unclear to me what backward compatibility guarantees _need_ to be maintained.  And perhaps I'm missing a use case. 
* I agree `dryrun` can be safely removed from the name.  Usage is qualified by the import.
* I'd consider one of the following options for `singleton` factory methods:
  * Do nothing and leave as is.
    * Personally, I'd prefer _not_ to include terms like _singleton_  in the method definition.  Though I don't feel strongly enough to pursue further.
  * If you're open to further renaming, I'd consider naming something specific to the functionality like:   `app_request_with_args`.
  * Delete the _singleton_ factory methods in favor of `build_dryrun_request`, _and_ add `args` factory methods to `App` and `LSig`.
    * If given a preference, I'd lean here.  
      * I think there's a large number of possible factory methods that _may_ exist for building a dryrun request.  I am thinking moving the factories one level up minimizes the surface area needing changes later.
      * Though I don't have enough experience using the repo to be confident about my thinking.

Reply:
I don't feel very strongly about naming. I do think it's convenient to have separate helper functions for **app** versus **logicsig** because it makes it very apparent to the end-user that they need to think this through. These helper functions are easy enough to maintain, as they route immediately to the `build_dryrun_request()` workhorse. It's worth soliciting more feedback/opinions.